mediators/datamapper/components/org.wso2.carbon.gateway.mediators.datamapper.engine/src/main/java/org/wso2/carbon/gateway/mediators/datamapper/engine/core/mapper/MappingHandler.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (c) 2016, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.carbon.gateway.mediators.datamapper.engine.core.mapper;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Map;
+
+import org.wso2.carbon.gateway.mediators.datamapper.engine.core.exceptions.JSException;
+import org.wso2.carbon.gateway.mediators.datamapper.engine.core.exceptions.ReaderException;
+import org.wso2.carbon.gateway.mediators.datamapper.engine.core.exceptions.SchemaException;
+import org.wso2.carbon.gateway.mediators.datamapper.engine.core.exceptions.WriterException;
+import org.wso2.carbon.gateway.mediators.datamapper.engine.core.executors.Executor;
+import org.wso2.carbon.gateway.mediators.datamapper.engine.core.executors.ScriptExecutorFactory;
+import org.wso2.carbon.gateway.mediators.datamapper.engine.core.models.Model;
+import org.wso2.carbon.gateway.mediators.datamapper.engine.core.notifiers.InputVariableNotifier;
+import org.wso2.carbon.gateway.mediators.datamapper.engine.core.notifiers.OutputVariableNotifier;
+import org.wso2.carbon.gateway.mediators.datamapper.engine.input.InputBuilder;
+import org.wso2.carbon.gateway.mediators.datamapper.engine.output.OutputMessageBuilder;
+import org.wso2.carbon.gateway.mediators.datamapper.engine.utils.InputOutputDataType;
+import org.wso2.carbon.gateway.mediators.datamapper.engine.utils.ModelType;
+
+/**
+ *  This class contains the methods necessary to transform a message.
+ */
+public class MappingHandler implements InputVariableNotifier, OutputVariableNotifier {
+
+    private int dmExecutorPoolSize;
+    private String inputVariable;
+    private String outputVariable;
```

Comment:
Documentation on these variables would be nice to have (adds lot of value to code)